.gitlab-ci.yml

```diff
@@ -1842,17 +1844,31 @@ deploy_windows_master-latest-7:
   script:
     - $S3_CP_CMD $OMNIBUS_PACKAGE_DIR/datadog-agent-7*-x86_64.msi "s3://$WINDOWS_BUILDS_S3_BUCKET/master/datadog-agent-7-latest.amd64.msi" --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
 
-# deploy windows packages to a public s3 bucket when tagged
-deploy_windows_tags:
+# triggered builds (6), deployed to bucket/tagged
+deploy_windows_tags-a6:
   stage: deploy6
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/deploy:$DATADOG_AGENT_BUILDERS
   before_script:
     - ls $OMNIBUS_PACKAGE_DIR
   <<: *run_when_triggered_on_tag_6
+  <<: *skip_when_unwanted_on_6
```

Comment:
No need for `skip_when_unwanted_on_6`. Putting both this and `run_when_triggered_on_tag_6`  in the same job results in an unexpected configuration, because we apply `run_when_triggered_on_tag_6` first:
```yaml
  only:
    refs:
      - triggers
  except:
    variables:
      - $RELEASE_VERSION_6 == "nightly"
      - $RELEASE_VERSION_6 == ""
```

then `skip_when_unwanted_on_6`:
```yaml
  except:
    variables:
      - $RELEASE_VERSION_6 == ""
```

And the result is:
```yaml
  only:
    refs:
      - triggers
  except:
    variables:
      - $RELEASE_VERSION_6 == ""
```
(because lists cannot be merged, so they're overriden).

In this case, if `$RELEASE_VERSION_6` is set to `nightly` (for example in the nightly jobs), the windows tags job will be executed, which is something we don't want to do.

Reply:
We have both in several places:
deploy_windows_tags-latest-6: https://github.com/DataDog/datadog-agent/blob/master/.gitlab-ci.yml#L1861
latest_release_6: https://github.com/DataDog/datadog-agent/blob/master/.gitlab-ci.yml#L2064
tag_release_6: https://github.com/DataDog/datadog-agent/blob/master/.gitlab-ci.yml#L2033

Should I remove it?

Reply:
Removed. Also merged the "update latest" for nightlies, since there we always run A6 and A7 together.